cmd/object-handlers.go

```diff
@@ -601,11 +601,14 @@ func (api objectAPIHandlers) CopyObjectPartHandler(w http.ResponseWriter, r *htt
 	var hrange *httpRange
 	rangeHeader := r.Header.Get("x-amz-copy-source-range")
 	if rangeHeader != "" {
-		if hrange, err = parseRequestRange(rangeHeader, objInfo.Size); err != nil {
+		if hrange, err = parseCopyPartRange(rangeHeader, objInfo.Size); err != nil {
 			// Handle only errInvalidRange
 			// Ignore other parse error and treat it as regular Get request like Amazon S3.
 			if err == errInvalidRange {
-				writeErrorResponse(w, ErrInvalidRange, r.URL)
+				writeErrorResponse(w, ErrInvalidCopyPartRange, r.URL)
+				return
+			} else if err == errInvalidRangeSource {
```

Comment:
For the sake of precaution, can you add an else which returns error unexpected ?